src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -312,7 +312,19 @@ moduleToJs (Module _ coms mn _ imps exps foreigns decls) foreign_ =
   qualifiedToJS f (Qualified _ a) = AST.Var Nothing $ identToJs (f a)
 
   foreignIdent :: Ident -> AST
-  foreignIdent ident = accessorString (mkString $ runIdent ident) (AST.Var Nothing "$foreign")
+  foreignIdent (runIdent -> unescaped)
+    | T.any (== '\'') unescaped
+    = (AST.Indexer Nothing
+        (AST.Ternary Nothing
+          (AST.App Nothing
+            (AST.Indexer Nothing (AST.StringLiteral Nothing "hasOwnProperty")
+                                 (AST.Var Nothing "$foreign"))
+            [AST.StringLiteral Nothing $ mkString unescaped])
+          (AST.StringLiteral Nothing $ mkString unescaped)
+          (AST.StringLiteral Nothing $ mkString (T.replace "'" "$prime" unescaped)))
+        (AST.Var Nothing "$foreign"))
+    | otherwise
+    = accessorString (mkString unescaped) (AST.Var Nothing "$foreign")
```

Comment:
Is this emitted for every reference to a foreign identifier within a module?

Reply:
`$foreign[$foreign.hasForeignProperty("foo'") ? "foo'" : "foo$prime"]` would have been emitted for every reference to a `foo'` identifier imported from a foreign module yes.